src/Centreon/Domain/Monitoring/Resource.php

```diff
@@ -0,0 +1,469 @@
+     */
+    public function getIcon(): ?Icon
+    {
+        return $this->icon;
+    }
+
+    /**
+     * @param \Centreon\Domain\Monitoring\Icon|null $icon
+     * @return \Centreon\Domain\Monitoring\Resource
+     */
+    public function setIcon(?Icon $icon): self
+    {
+        $this->icon = $icon;
+
+        return $this;
+    }
+
+    /**
+     * @return \Centreon\Domain\Monitoring\Resource|null
+     */
+    public function getParent(): ?Resource
+    {
+        return $this->parent;
+    }
+
+    /**
+     * @param \Centreon\Domain\Monitoring\Resource|null $parent
+     * @return \Centreon\Domain\Monitoring\Resource
+     */
+    public function setParent(?Resource $parent): self
+    {
+        $this->parent = $parent;
+
+        return $this;
+    }
+
+    /**
+     * @return \Centreon\Domain\Monitoring\ResourceStatus|null
+     */
+    public function getStatus(): ?ResourceStatus
+    {
+        return $this->status;
+    }
+
+    /**
+     * @param \Centreon\Domain\Monitoring\ResourceStatus|null $status
+     * @return \Centreon\Domain\Monitoring\Resource
+     */
+    public function setStatus(?ResourceStatus $status): self
+    {
+        $this->status = $status;
+
+        return $this;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function getInDowntime(): ?bool
+    {
+        return $this->inDowntime;
+    }
+
+    /**
+     * @param bool|null $inDowntime
+     * @return \Centreon\Domain\Monitoring\Resource
+     */
+    public function setInDowntime(?bool $inDowntime): self
+    {
+        $this->inDowntime = $inDowntime;
+
+        return $this;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function getAcknowledged(): ?bool
+    {
+        return $this->acknowledged;
+    }
+
+    /**
+     * @param bool|null $acknowledged
+     * @return \Centreon\Domain\Monitoring\Resource
+     */
+    public function setAcknowledged(?bool $acknowledged): self
+    {
+        $this->acknowledged = $acknowledged;
+
+        return $this;
+    }
+
+    /**
+     * @return \Centreon\Domain\Monitoring\ResourceSeverity|null
+     */
+    public function getSeverity(): ?ResourceSeverity
+    {
+        return $this->severity;
+    }
+
+    /**
+     * @param \Centreon\Domain\Monitoring\ResourceSeverity|null $severity
+     * @return \Centreon\Domain\Monitoring\Resource
+     */
+    public function setSeverity(?ResourceSeverity $severity): self
+    {
+        $this->severity = $severity;
+
+        return $this;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getImpactedResourcesCount(): ?int
+    {
+        return $this->impactedResourcesCount;
+    }
+
+    /**
+     * @param int|null $impactedResourcesCount
+     * @return \Centreon\Domain\Monitoring\Resource
+     */
+    public function setImpactedResourcesCount(?int $impactedResourcesCount): self
+    {
+        $this->impactedResourcesCount = $impactedResourcesCount ?: 0;
```

Comment:
Might as well not allow a null value. Set the default value to 0.

Reply:
where you suggest the default value to be set?

Reply:
Directly when defining the property 
**private $impactedResourcesCount = 0;**